tests/test_transform_parametrise.py

```diff
@@ -0,0 +1,477 @@
+    """
+    Check the parameters, subroutine and call arguments of each subroutine.
+    """
+    item = scheduler.item_map['parametrise#driver']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["driver"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["driver"]
+    for call in FindNodes(ir.CallStatement).visit(item.routine.body):
+        if "kernel1" in call.name:
+            assert call.arguments == call_arguments["kernel1"]
+        elif "kernel2" in call.name:
+            assert call.arguments == call_arguments["kernel2"]
+    item = scheduler.item_map['parametrise#another_driver']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["another_driver"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["another_driver"]
+    for call in FindNodes(ir.CallStatement).visit(item.routine.body):
+        if "kernel1" in call.name:
+            assert call.arguments == call_arguments["kernel1"]
+    item = scheduler.item_map['parametrise#kernel1']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["kernel1"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["kernel1"]
+    item = scheduler.item_map['parametrise#kernel2']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["kernel2"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["kernel2"]
+    for call in FindNodes(ir.CallStatement).visit(item.routine.body):
+        if "device1" in call.name:
+            assert call.arguments == call_arguments["device1"]
+        elif "device2" in call.name:
+            assert call.arguments == call_arguments["device2"]
+    item = scheduler.item_map['parametrise#device1']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["device1"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["device1"]
+    for call in FindNodes(ir.CallStatement).visit(item.routine.body):
+        if "device2" in call.name:
+            assert call.arguments == call_arguments["device2"]
+    item = scheduler.item_map['parametrise#device2']
+    routine_parameters = [var for var in item.routine.variables if var.type.parameter]
+    assert routine_parameters == parameter_variables["device2"]
+    assert [arg.name for arg in item.routine.arguments] == subroutine_arguments["device2"]
+
+
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_source(here, frontend, config):
```

Comment:
If you don't mind, may I suggest a little more verbose test names? That makes it easier to select them directly via something like `pytest -k parametrise_source tests` (which I do a lot), whereas `test_source` may easily get ambiguous`.